libres/lib/analysis/enkf_linalg.cpp

```diff
@@ -93,60 +93,55 @@ void enkf_linalg_genX2(matrix_type *X2, const matrix_type *S,
 /*This function is similar to enkf_linalg_svdS but it returns the eigen values without its inverse and also give the matrices truncated U VT and Sig0*/
 
 // Trunc.SVD(S)  = U0 * Sig0 * V0T
-int enkf_linalg_svd_truncation(const matrix_type *S, double truncation,
-                               int ncomp, dgesvd_vector_enum store_V0T,
-                               double *sig0, matrix_type *U0,
-                               matrix_type *V0T) {
+int enkf_linalg_svd_truncation(const matrix_type *S,
+                               const std::variant<double, int> &truncation,
+                               dgesvd_vector_enum store_V0T, double *sig0,
+                               matrix_type *U0, matrix_type *V0T) {
 
     int num_significant = -1;
     int nrows = matrix_get_rows(S);
     int ncolumns = matrix_get_columns(S);
 
-    if (((truncation > 0) && (ncomp < 0)) ||
```

Comment:
Are we comfortable removing this check?

Reply:
Yes - the point about using `std::variant<>` is to enforce this from the compiler

Reply:
I see, bit I don't think there is anything stopping us from passing in a negative truncation.
Perhaps this is a non-issue?

Reply:
> I see, bit I don't think there is anything stopping us from passing in a negative truncation.

No - in principle I agree with that, but that should caught when the truncation is set.

Reply:
I agree.